py/surveyqa/nightly.py

```diff
@@ -387,3 +447,73 @@ def makeplots(night, exposures, tiles, outdir):
         fx.write(html)
 
     print('Wrote {}'.format(outfile))
+
+<<<<<<< HEAD
+
+=======
+>>>>>>> 2740d584a08dd6f2e9054ea0c4833525bd611a9e
+def get_exptype_counts(exposures, calibs, width=300, height=300):
+    """
+    Generate a horizontal bar plot showing the counts for each type of exposure grouped 
+    by whether they have FLAVOR='science' or PROGRAM='calib'
+    
+    ARGS:
+        exposures : a table of exposures which only contain those with FLAVOR='science'
+        calibs : a table of exposures which only contains those with PROGRAm='calibs'
+    """
+    darks = len(exposures[exposures['PROGRAM'] == 'DARK'])
+    grays = len(exposures[exposures['PROGRAM'] == 'GRAY'])
+    brights = len(exposures[exposures['PROGRAM'] == 'BRIGHTS'])
+    
+    arcs = len(calibs[calibs['FLAVOR'] == 'arc'])
+    flats = len(calibs[calibs['FLAVOR'] == 'flat'])
+    zeroes = len(calibs[calibs['FLAVOR'] == 'zero'])
+    
+    types = [('calib', 'ZERO'), ('calib', 'FLAT'), ('calib', 'ARC'), 
+            ('science', 'BRIGHT'), ('science', 'GRAY'), ('science', 'DARK')]
+    counts = np.array([zeroes, flats, arcs, brights, grays, darks])
+    
+    src = ColumnDataSource({'types':types, 'counts':counts})
+    
+    p = bk.figure(width=width, height=height,
+                  y_range=FactorRange(*types), title='Exposure Type Counts', 
+                  toolbar_location=None)
+    p.hbar(y='types', right='counts', left=0, height=0.5, line_color='white',
+           fill_color=factor_cmap('types', palette=Spectral6, factors=types), source=src)
+    
+    
+    labels = LabelSet(x='counts', y='types', text='counts', level='glyph', source=src, 
+                      render_mode='canvas', x_offset=5, y_offset=-10, text_color='gray', text_font='sans-serif')
+    p.add_layout(labels)
+    
+    p.ygrid.grid_line_color=None
+    
+    return p
+
+def get_night_link(night, exposures):
+    '''Gets the href string for the previous night and the next night for a given nightly page. 
+        Input:
+            night: night value, string
+            exposures: Table with columns...
+        Output: [previous page href, next page href], elements are strings'''
+    unique_nights = list(np.unique(exposures['NIGHT'])) 
+    ind = unique_nights.index(night)
+    
+    if ind == 0:
+        prev_night = night
+        next_night = unique_nights[ind+1]
+    
+    if ind == (len(unique_nights)-1):
+        prev_night = unique_nights[ind-1]
+        next_night = night
+        
+    if ind != 0 and ind != (len(unique_nights)-1):
+        prev_night = unique_nights[ind-1]
+        next_night = unique_nights[ind+1]
+    
+    outdir = os.path.join(os.getcwd(), 'survey-qa')
+    os.makedirs(outdir, exist_ok=True)
+    prev_str = "file://" + str(outdir) + "/night-{}.html".format(prev_night)
+    next_str = "file://" + str(outdir) + "/night-{}.html".format(next_night)
```

Comment:
See comment above about dropping the `"file://" + str(outdir)` part; just give local `"night-{}.html".format(next_night)` (NOTE: without the preceding slash '/').